### DISCUSSION TITLE
How can I update/upgrade OhMyPosh to the latest version?

### DISCUSSION THREAD

**Discussion by Pyrrhusn** [JanDeDobbeleer/oh-my-posh#1219] [Troubleshoot]
Sorry if this question's already been asked but I couldn't find about it. I want to update OhMyPosh to the latest version, currently running version 5.16.4 on ubuntu wsl2. I had installed omp using the linux installtion because i was running ubuntu terminal on wsl2, is that fuine or should I reinstall it using the windows installation procedure. What I was looking for was a command or a way to update my omp setup, like how you can on oh my zsh wit omz update. Some further clarification I'm using zsh. Thank you for the help.
Sorry again if this is a fumb question and already questioned/answered.

**Comment by JanDeDobbeleer** [comment]
You should be able to run the installation instructions again (always fetches latest):

```bash
sudo wget https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/posh-linux-amd64 -O /usr/local/bin/oh-my-posh
sudo chmod +x /usr/local/bin/oh-my-posh
```

For convenience, you can always [install](https://ohmyposh.dev/docs/windows#installation) using Winget, that way you have the `oh-my-posh-wsl` executable in your `PATH` on unix. simply delete the current `oh-my-posh` binary you have, install on Windows with Winget and replace `oh-my-posh` with `oh-my-posh-wsl` in Ubuntu.

**Reply by Pyrrhusn** [reply]
Thanks for the suggestion. I manually removed the old oh-my-posh stuff such as the themes folder in the root and the omp from  the usr/local folder. After that I installed omp again using Winget on powershell and followed the steps for zsh. But now I have another problems that came up because, I think, of how I uninstalled/installed omp (again). I'm afraid I didn't do a clean uninstall and/or installation. The prompts are too slow, taking longer 200ms where as before they were very quick and snappy, most of time around the 20ms mark. So what should I do now? Is there a way to completely uninstall/remove everything related to oh-my-posh and do full clean reinstall? Thanks again

**Reply by JanDeDobbeleer** [reply]
Did you, in WSL, replace `oh-m-posh` with `oh-my-posh-wsl`?

**Reply by Pyrrhusn** [reply]
Yup I did, double-checked

**Reply by JanDeDobbeleer** [reply]
Can you do me a favor, move that executable to the WSL mount (same location where you initially installed the original executable).

**Reply by Pyrrhusn** [reply]
<img width="866" alt="Screenshot 2021-11-14 123043" src="https://user-images.githubusercontent.com/43817108/141679188-3e1595d3-05fd-4877-a216-09e28cbf9aca.png">
Running the --debug option gives this, the latency on the prompts is the same as before, almost negligible, but the latency between prompts, them reappearing and the execution time is unbearable, it shouldn't be 1 sec. Could it be that oh-my-posh-wsl is being slowed down by something? 

<img width="533" alt="Screenshot 2021-11-14 124934" src="https://user-images.githubusercontent.com/43817108/141679551-60a847a6-32b0-44b9-83ed-c4c5c70a43df.png">
This is where the command `winget install JanDeDobbeleer.OhMyPosh` installed oh-my-posh. The WSL and the oher are in the same file, but the wsl version is not an executable/application, its just a file. Is that the problem? 
Another thing, the winget installation command installs omp for me in the AppData... and later on the page it says `eval "$(oh-my-posh --init --shell zsh --config ~/.poshthemes/jandedobbeleer.omp.json)"` but this can't work because there is no .poshthemes folder in the root, unless installed using the Linux process.  So I had to manually make one to make the themes work, but for some reason the prompts are jus slow. What's going on then? 


Could you also specify this "move that executable to the WSL mount (same location where you initially installed the original executable)."?

Edit: An interesting thing I found out, whenever my laptop's charging the latency drops down to below the 200ms mark but this behavior is very weird. It shouldn't work like that, right?

If this doesn't get solved out, I can just install it again with the Linux process as I did before, and It'd work brilliantly.
